org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/StandardPreferenceManager.java

```diff
@@ -83,25 +84,40 @@ public static void initializeMavenPreferences() {
 	public void update(Preferences preferences) {
 		super.update(preferences);
 
+		boolean updateMavenProjects = false;
 		String newMavenSettings = preferences.getMavenUserSettings();
 		String oldMavenSettings = getMavenConfiguration().getUserSettingsFile();
 		if (!Objects.equals(newMavenSettings, oldMavenSettings)) {
 			try {
 				getMavenConfiguration().setUserSettingsFile(newMavenSettings);
+				updateMavenProjects = true;
 			} catch (CoreException e) {
 				JavaLanguageServerPlugin.logException("failed to set Maven settings", e);
 				preferences.setMavenUserSettings(oldMavenSettings);
 			}
 		}
-		boolean updateMavenProjects = false;
 		String newMavenGlobalSettings = preferences.getMavenGlobalSettings();
 		String oldMavenGlobalSettings = getMavenConfiguration().getGlobalSettingsFile();
 		if (!Objects.equals(newMavenGlobalSettings, oldMavenGlobalSettings)) {
 			try {
 				getMavenConfiguration().setGlobalSettingsFile(newMavenGlobalSettings);
+				updateMavenProjects = true;
 			} catch (CoreException e) {
 				JavaLanguageServerPlugin.logException("failed to set Maven global settings", e);
-				preferences.setMavenUserSettings(oldMavenSettings);
+				preferences.setMavenGlobalSettings(oldMavenGlobalSettings);
+			}
+		}
+		String newMavenLifecycleMappings = preferences.getMavenLifecycleMappings();
+		IEclipsePreferences prefs = InstanceScope.INSTANCE.getNode(IMavenConstants.PLUGIN_ID);
+		String oldMavenLifecycleMappings = prefs.get(MavenPreferenceConstants.P_WORKSPACE_MAPPINGS_LOCATION, null);
```

Comment:
Why not to use its getter api `getMavenConfiguration().getWorkspaceLifecycleMappingMetadataFile()` to retrieve the current lifecycle mapping config? I see you have used its setter api to update the lifecycle mapping config in subsequent operation.

Reply:
We don't want to update the projects if the maven lifecycle mapping is not set.
m2e lifecycle mapping is `<m2e_core_location>/lifecycle-mapping-metadata.xml` by default, VS Code lifecycle mapping is null by default
See https://github.com/eclipse-m2e/m2e-core/blob/db42af86c7976a6131daf4cda1a10857e497bf92/org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/preferences/MavenConfigurationImpl.java#L266

Reply:
This touches too much the internal implementation of m2e, which is not a good practice from a maintenance standpoint.

To identify if users have changed the lifecycle mapping in jdt.ls, I prefer to store the lifecycle mapping setting within jdt.ls metadata cache. We can then use jdt.ls preference cache to identify any diff. If any modification is detected, then call m2e API to apply it.

Reply:
trying...